From 4d2647608289391e9ab2bb71e87f793ff418bd7e Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Mon, 3 Mar 2014 16:31:13 +1100
Subject: [PATCH] Revert "audioflinger: Power hints for audio activity"

This reverts commit ff7ae18ca39ced274f37856b8da9dc1e446f69dd.

diff --git a/services/audioflinger/AudioPolicyService.cpp b/services/audioflinger/AudioPolicyService.cpp
index 94f4ab7..ccd14c4 100644
--- a/services/audioflinger/AudioPolicyService.cpp
+++ b/services/audioflinger/AudioPolicyService.cpp
@@ -96,9 +96,6 @@ AudioPolicyService::AudioPolicyService()
     if (rc)
         return;
 
-    rc = hw_get_module(POWER_HARDWARE_MODULE_ID, (const hw_module_t **)&mPowerModule);
-    ALOGW_IF(rc, "couldn't get power module (%s)", strerror(-rc));
-
     ALOGI("Loaded audio policy from %s (%s)", module->name, module->id);
 
     // load audio pre processing modules
@@ -249,7 +246,6 @@ status_t AudioPolicyService::startOutput(audio_io_handle_t output,
     }
     ALOGV("startOutput()");
     Mutex::Autolock _l(mLock);
-    setPowerHint(true);
     return mpAudioPolicy->start_output(mpAudioPolicy, output, stream, session);
 }
 
@@ -271,9 +267,7 @@ status_t  AudioPolicyService::doStopOutput(audio_io_handle_t output,
 {
     ALOGV("doStopOutput from tid %d", gettid());
     Mutex::Autolock _l(mLock);
-    status_t ret = mpAudioPolicy->stop_output(mpAudioPolicy, output, stream, session);
-    setPowerHint(false);
-    return ret;
+    return mpAudioPolicy->stop_output(mpAudioPolicy, output, stream, session);
 }
 
 void AudioPolicyService::releaseOutput(audio_io_handle_t output)
@@ -360,7 +354,7 @@ status_t AudioPolicyService::startInput(audio_io_handle_t input)
         return NO_INIT;
     }
     Mutex::Autolock _l(mLock);
-    setPowerHint(true);
+
     return mpAudioPolicy->start_input(mpAudioPolicy, input);
 }
 
@@ -371,9 +365,7 @@ status_t AudioPolicyService::stopInput(audio_io_handle_t input)
     }
     Mutex::Autolock _l(mLock);
 
-    status_t ret = mpAudioPolicy->stop_input(mpAudioPolicy, input);
-    setPowerHint(false);
-    return ret;
+    return mpAudioPolicy->stop_input(mpAudioPolicy, input);
 }
 
 void AudioPolicyService::releaseInput(audio_io_handle_t input)
@@ -592,13 +584,6 @@ status_t AudioPolicyService::queryDefaultPreProcessing(int audioSession,
     return status;
 }
 
-void AudioPolicyService::setPowerHint(bool active) {
-    if (mPowerModule && mPowerModule->powerHint) {
-        mPowerModule->powerHint(mPowerModule, POWER_HINT_AUDIO,
-                active ? (void *)"state=1" : (void *)"state=0");
-    }
-}
-
 void AudioPolicyService::binderDied(const wp<IBinder>& who) {
     ALOGW("binderDied() %p, calling pid %d", who.unsafe_get(),
             IPCThreadState::self()->getCallingPid());
diff --git a/services/audioflinger/AudioPolicyService.h b/services/audioflinger/AudioPolicyService.h
index 577688b..ae053a9 100644
--- a/services/audioflinger/AudioPolicyService.h
+++ b/services/audioflinger/AudioPolicyService.h
@@ -27,7 +27,6 @@
 #include <system/audio.h>
 #include <system/audio_policy.h>
 #include <hardware/audio_policy.h>
-#include <hardware/power.h>
 #include <media/IAudioPolicyService.h>
 #include <media/ToneGenerator.h>
 #include <media/AudioEffect.h>
@@ -337,7 +336,6 @@ private:
     // Internal dump utilities.
     status_t dumpPermissionDenial(int fd);
 
-    void setPowerHint(bool active);
 
     mutable Mutex mLock;    // prevents concurrent access to AudioPolicy manager functions changing
                             // device connection state  or routing
@@ -348,8 +346,6 @@ private:
     struct audio_policy *mpAudioPolicy;
     KeyedVector< audio_source_t, InputSourceDesc* > mInputSources;
     KeyedVector< audio_io_handle_t, InputDesc* > mInputs;
-
-    power_module_t *mPowerModule;
 };
 
 }; // namespace android
